### PR TITLE
Fixing debugging duplicates. Adding a flag nondex.printstack to commu…

### PR DESCRIPTION
**Pull request by alexgyori**
…nicate to the runner that the stack needs to be printed when debugging is done. this flag is added to configuration.

### CODE REVIEW

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java [hunk 1]
```diff
@@ -58,13 +58,13 @@ public NonDexSurefireExecution(Mode mode, int seed, Pattern filter, long start,
                 this.executionId);
     }
 
-    public NonDexSurefireExecution(Configuration config, int start, int end, String test, Plugin surefire,
+    public NonDexSurefireExecution(Configuration config, long start, long end, boolean print, String test, Plugin surefire,
```
Comment (alexgyori) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2745242717/issues/source?bid=3322087&fileBranchId=3356590#l61)

File: nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java [hunk 2]
```diff
@@ -71,6 +73,11 @@ protected Configuration(Mode mode, int seed, Pattern filter, String executionId)
 
     public Configuration(Mode mode, int seed, Pattern filter, long start, long end, String nondexDir,
             String nondexJarDir, String testName, String executionId) {
+        this(mode, seed, filter, start, end, nondexDir, nondexJarDir, testName, executionId, false);
+    }
+
+    public Configuration(Mode mode, int seed, Pattern filter, long start, long end, String nondexDir,
```
Comment (alexgyori) [inline nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2745242680/issues/source?bid=3322087&fileBranchId=3356590#l79)

**Comment by august782** [comment]
NPE can occur at line 161 of DebugTask.java, as if there is no failure, failWithConfig returns null, so if the debugging fails, it would thrown an NPE.

**Comment by alexgyori** [comment]
I think this should fix it.

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java (hunk 1 above)
Comment (alexgyori) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2745399835/issues/source?bid=3322087&fileBranchId=3356590#l61)

File: nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java (hunk 2 above)
Comment (alexgyori) [inline nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2745399763/issues/source?bid=3322087&fileBranchId=3356590#l79)

**Comment by alexgyori** [comment]
The issue with the multiple debug files is due to the fact that you may end up running 17-17 18-18 as two different runs with equal limits, one failing and the other not failing, so you will have a debug file that prints the stack trace when the test does not fail.

File: nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java (hunk 2 above)
Comment (alexgyori) [inline nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747718196/issues/source?bid=3322087&fileBranchId=3356590#l79)

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java (hunk 1 above)
Comment (alexgyori) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747718233/issues/source?bid=3322087&fileBranchId=3356590#l61)

Comment (alexgyori) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747736738/issues/source?bid=3322087&fileBranchId=3356590#l61)

File: nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java (hunk 2 above)
Comment (alexgyori) [inline nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747736701/issues/source?bid=3322087&fileBranchId=3356590#l79)

File: nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java (hunk 1 above)
Comment (alexgyori) [inline nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747802802/issues/source?bid=3322087&fileBranchId=3356590#l61)

File: nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java (hunk 2 above)
Comment (alexgyori) [inline nondex-common/src/main/java/edu/illinois/nondex/common/Configuration.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/gyori/NonDex/file/2747802765/issues/source?bid=3322087&fileBranchId=3356590#l79)

**Pull request merged**